### PR TITLE
ref(starfish): Split up span summary page components

### PR DESCRIPTION
**Pull request by gggritso**
This one ended up bigger than I hoped, since it's such a ball of yarn. Splitting up the huge span summary view component so I can use the pieces separately when I move things out of Starfish. This is more of a setup for later work.

Also clarified some weirdly complicated functions, and fixed a couple of typos.

### CODE REVIEW

File: static/app/views/starfish/views/spanSummaryPage/index.tsx [hunk 1]
```diff
@@ -83,49 +66,21 @@ function SpanSummaryPage({params, location}: Props) {
...
-      : 'Requests',
-    data: spanMetricsSeriesData?.['spm()'].data,
   };
 
-  const title = getDescriptionLabel(span[SpanMetricsFields.SPAN_OP], true);
-  const spanDescriptionCardTitle = getDescriptionLabel(span[SpanMetricsFields.SPAN_OP]);
+  const title = [
+    getSpanOperationDescription(span[SpanMetricsFields.SPAN_OP]),
+    t('Summary'),
+  ].join(' ');
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/spanSummaryPage/index.tsx:83]: l: Why the array over a templated string?

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+};
+
+export function SpanSummaryView({groupId}: Props) {
+  const location = useLocation<Query>();
+  const {endpoint, endpointMethod} = location.query;
+
+  const {data: fullSpan} = useFullSpanFromTrace(groupId);
+
+  const queryFilter: SpanSummaryQueryFilters = endpoint
+    ? {transactionName: endpoint, 'transaction.method': endpointMethod}
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:53]: l: This sets `transaction.method` to undefined if `endpointMethod` is not defined? Is that fine?

Can we avoid having the duplicate setters for `queryFilter['transaction.method']`?

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+      'api.starfish.span-summary-page-metrics-chart'
+    );
+
+  useSynchronizeCharts([!areSpanMetricsSeriesLoading]);
+
+  const spanMetricsThroughputSeries = {
+    seriesName: span?.[SpanMetricsFields.SPAN_OP]?.startsWith('db')
+      ? 'Queries'
+      : 'Requests',
+    data: spanMetricsSeriesData?.['spm()'].data,
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:103]: l: Can we avoid hardcoding `spm()` `http_error_count()` and others?

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+
+  const span = {
+    ...spanMetrics,
+    [SpanMetricsFields.SPAN_GROUP]: groupId,
+  } as {
+    [SpanMetricsFields.SPAN_OP]: string;
+    [SpanMetricsFields.SPAN_DESCRIPTION]: string;
+    [SpanMetricsFields.SPAN_ACTION]: string;
+    [SpanMetricsFields.SPAN_DOMAIN]: string;
+    [SpanMetricsFields.SPAN_GROUP]: string;
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:86]: these should be optional right?

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+        </FilterOptionsContainer>
+
+        <BlockContainer>
+          {span?.[SpanMetricsFields.SPAN_OP]?.startsWith('db') &&
+            span?.[SpanMetricsFields.SPAN_OP] !== 'db.redis' && (
+              <Block title={t('Table')}>{span?.[SpanMetricsFields.SPAN_DOMAIN]}</Block>
+            )}
+
+          <Block
+            title={getThroughputTitle(span?.[SpanMetricsFields.SPAN_OP])}
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:124]: feels weird to have all these optional chaining operators around here w/ span, but maybe we can fix this afterwards with a bigger types refactor.

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+              total={spanMetrics?.[`sum(${SpanMetricsFields.SPAN_SELF_TIME})`]}
+            />
+          </Block>
+        </BlockContainer>
+      </HeaderContainer>
+
+      {span?.[SpanMetricsFields.SPAN_DESCRIPTION] && (
+        <DescriptionContainer>
+          <SpanDescription
+            span={{
```
Comment (AbhiPrasad) [inline static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:174]: I wonder if `SpanDescription` should even get access to the entire span - it really only cares about a description string.

**Review by AbhiPrasad — APPROVED** [review]
Good refactor!

Any feedback on `static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx` should probably be looked at in a follow up PR.

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx (hunk 6 above)
Comment (gggritso) [reply static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:174]: It definitely should not. It cares about description and op, and it should take them separately. The typing specifies that it only needs a subset, but that's not convenient or clear!

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx (hunk 5 above)
Comment (gggritso) [reply static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:124]: Agreed, though when `span` comes from the network, there are lot of quirky places where information might be partially unavailable. We should clarify this regardless, though

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx (hunk 4 above)
Comment (gggritso) [reply static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:86]: Definitely yes, I just didn't have a moment to fix this

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx (hunk 3 above)
Comment (gggritso) [reply static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:103]: Done!

File: static/app/views/starfish/views/spanSummaryPage/index.tsx (hunk 1 above)
Comment (gggritso) [reply static/app/views/starfish/views/spanSummaryPage/index.tsx:83]: Saw this pattern elsewhere in Performance and used it without thinking too deeply, TBH!

File: static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx (hunk 2 above)
Comment (gggritso) [reply static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx:53]: Yep, it's fine if that happens! No idea why the duplicate setter exists, removed

**Pull request merged**